crawler/regelleistung.py

```diff
@@ -148,62 +432,53 @@ def get_df_for_date(url, date_to_get):
-    hours_diff = hours_to_int - hours_from_int
-    cols_to_adapt = [
-        "total_min_capacity_price_eur_mw",
-        "total_average_capacity_price_eur_mw",
-        "total_marginal_capacity_price_eur_mw",
-        "germany_min_capacity_price_eur_mw",
-        "germany_average_capacity_price_eur_mw",
-        "germany_marginal_capacity_price_eur_mw",
-        "austria_min_capacity_price_eur_mw",
-        "austria_average_capacity_price_eur_mw",
-        "austria_marginal_capacity_price_eur_mw",
-        "capacity_price_eur_mw",
-    ]
-    for col_to_adapt in cols_to_adapt:
-        if col_to_adapt in df.columns:
-            final_col_name = col_to_adapt + "h"
-            df[final_col_name] = pd.to_numeric(df[col_to_adapt]) / hours_diff
-
-    cols_to_drop = [
-        "total_min_energy_price_eur_mwh",
-        "total_average_energy_price_eur_mwh",
-        "total_marginal_energy_price_eur_mwh",
-        "germany_min_energy_price_eur_mwh",
-        "germany_average_energy_price_eur_mwh",
-        "germany_marginal_energy_price_eur_mwh",
-        "austria_min_energy_price_eur_mwh",
-        "austria_average_energy_price_eur_mwh",
-        "austria_marginal_energy_price_eur_mwh",
-        "total_min_capacity_price_eur_mw",
-        "total_average_capacity_price_eur_mw",
-        "total_marginal_capacity_price_eur_mw",
-        "germany_min_capacity_price_eur_mw",
-        "germany_average_capacity_price_eur_mw",
-        "germany_marginal_capacity_price_eur_mw",
-        "austria_min_capacity_price_eur_mw",
-        "austria_average_capacity_price_eur_mw",
-        "austria_marginal_capacity_price_eur_mw",
-        "capacity_price_eur_mw",
-    ]
-    for col_to_drop in cols_to_drop:
-        if col_to_drop in df.columns:
-            df = df.drop(col_to_drop, axis=1)
+    if get_date_column_from_table_name(table_name) == "date_from":
+        hours_from = np.array([product_list[1] for product_list in product_split_array])
+        hours_to = np.array([product_list[2] for product_list in product_split_array])
+        hours_from_int = (hours_from).astype(np.int16)
+        hours_to_int = (hours_to).astype(np.int16)
+        hours_diff = hours_to_int - hours_from_int
+        cols_to_adapt = [
+            "total_min_capacity_price_eur_mw",
+            "total_average_capacity_price_eur_mw",
+            "total_marginal_capacity_price_eur_mw",
+            "germany_min_capacity_price_eur_mw",
+            "germany_average_capacity_price_eur_mw",
+            "germany_marginal_capacity_price_eur_mw",
+            "austria_min_capacity_price_eur_mw",
+            "austria_average_capacity_price_eur_mw",
+            "austria_marginal_capacity_price_eur_mw",
+            "capacity_price_eur_mw",
+        ]
+        for col_to_adapt in cols_to_adapt:
+            if col_to_adapt in df.columns:
+                final_col_name = col_to_adapt + "h"
+                df[final_col_name] = pd.to_numeric(df[col_to_adapt]) / hours_diff
```

Comment:
so for the MW values the price was much more than the price per MWh right?

e.g. 100€/MW is 1MW for 4 Hours, so the actual price per MWH is 25€/MWh - seems correct.

dividing this gives a smooth curve since the transition?

Reply:
so for the MW values the price was much more than the price per MWh right? -> yes